File: src/main/java/com/dh/ondot/schedule/domain/service/PlaceHistoryService.java
package com.dh.ondot.schedule.domain.service;

import com.dh.ondot.core.util.TimeUtils;
import com.dh.ondot.schedule.domain.PlaceHistory;
import com.dh.ondot.schedule.infra.redis.PlaceHistoryRedisRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.time.Instant;
import java.time.LocalDateTime;
import java.util.List;

@Service
@RequiredArgsConstructor
public class PlaceHistoryService {

    private final PlaceHistoryRedisRepository repository;

    public void record(
            Long memberId, String title, String roadAddress,
            Double lon, Double lat
    ) {
        String finalTitle = (title == null || title.isBlank())
                ? roadAddress
                : title;

        repository.push(PlaceHistory.of(memberId, finalTitle, roadAddress, lon, lat));
    }

    public List<PlaceHistory> recent(Long memberId) {
        return repository.findRecent(memberId);
    }

    public void delete(Long memberId, LocalDateTime searchedAt) {
        Instant instant = TimeUtils.toInstant(searchedAt);
        repository.removeByTimestamp(memberId, instant);
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/domain/service/PlaceHistoryService.kt
package com.dh.ondot.schedule.domain.service

import com.dh.ondot.core.util.TimeUtils
import com.dh.ondot.schedule.domain.PlaceHistory
import com.dh.ondot.schedule.infra.redis.PlaceHistoryRedisRepository
import org.springframework.stereotype.Service
import java.time.LocalDateTime

@Service
class PlaceHistoryService(
    private val repository: PlaceHistoryRedisRepository,
) {
    fun record(
        memberId: Long, title: String?, roadAddress: String,
        lon: Double, lat: Double,
    ) {
        val finalTitle = if (title.isNullOrBlank()) roadAddress else title

        repository.push(PlaceHistory.of(memberId, finalTitle, roadAddress, lon, lat))
    }

    fun recent(memberId: Long): List<PlaceHistory> =
        repository.findRecent(memberId)

    fun delete(memberId: Long, searchedAt: LocalDateTime) {
        val instant = TimeUtils.toInstant(searchedAt)
        repository.removeByTimestamp(memberId, instant)
    }
}
